plugins/dicom_viewer/server/__init__.py

```diff
@@ -206,19 +207,24 @@ def dicomSubstringSearchHandler(query, types, user=None, level=None, limit=0, of
     """
     if types != ['item']:
         raise RestException('The dicom search is only able to search in Item.')
+    if not isinstance(query, six.string_types):
+        raise RestException('The search query must be a string.')
 
     jsQuery = """
         function() {
-            var queryKey = '%(queryKey)s'.toLowerCase();
-            var queryValue = '%(queryValue)s'.toLowerCase();
+            var queryKey = %(query)s.toLowerCase();
+            var queryValue = queryKey;
             var dicomMeta = obj.dicom.meta;
             return Object.keys(dicomMeta).some(
                 function(key) {
                     return (key.toLowerCase().indexOf(queryKey) !== -1)  ||
                         dicomMeta[key].toString().toLowerCase().indexOf(queryValue) !== -1;
                 })
             }
-        """ % {'queryKey': query, 'queryValue': query}
+    """ % {
+        # This could eventually be a separately-defined key and value
+        'query': json.dumps(query)
```

Comment:
The `dumps` serialization will escape any backslashes and output a JSON-safe string (which is the same as a Javascript string literal).